tests/test_pegen.py

```diff
@@ -556,3 +556,14 @@ def test_missing_start() -> None:
     """
     with pytest.raises(GrammarError):
         parser_class = make_parser(grammar)
+
+
+def test_soft_keyword() -> None:
```

Comment:
I am missing how this test is testing the `!(SOFT_KEYWORD)` part. As `"number" n=NUMBER` anything using `"number" NUMBER` will match the first rule without even testing the second one. You need to revert the order at least so the  `!(SOFT_KEYWORD)` is actually covered when you pass `"number 1"`, no?

I am missing something?

Reply:
I think you are right. I will double check.

Reply:
I think a better test would be to have a bunch of rules with soft keywords and then have one that accepts SOFT_KEYWORD and something else and try that one with all the other soft keywords

Reply:
I did something along those lines. Let me know what you think.

Reply:
Beautiful